pkg/plugin/util/util.go

```diff
@@ -110,3 +116,67 @@ func SelfLinkToCRDName(selfLink string) string {
 		return components[pluralIndex]
 	}
 }
+
+func RetrieveStorageClassMapping(config *rest.Config, veleroNs string, storageClassMapping map[string]string, logger logrus.FieldLogger) error {
+	var err error
+	if config == nil {
+		config, err = rest.InClusterConfig()
+		if err != nil {
+			logger.WithError(err).Errorf("Failed to get k8s inClusterConfig")
+			return errors.Wrap(err, "Could not retrieve in-cluster config")
+		}
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		logger.WithError(err).Errorf("Failed to get k8s clientset from the given config: %v", config)
+		return err
+	}
+	labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{"velero.io/plugin-config":"", "velero.io/change-storage-class":"RestoreItemAction"}}
+	configMaps, err := clientset.CoreV1().ConfigMaps(veleroNs).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
+	})
+	if err != nil {
+		logger.WithError(err).Errorf("Failed to retrieve config map for RestoreItemAction")
+		return err
+	}
+	for _, configMap := range configMaps.Items {
+		data := configMap.Data
+		for old, new := range data {
+			if exist, ok := storageClassMapping[old]; ok {
+				errMsg := fmt.Sprintf("Old storage class name %s has existing mapping new class storage class name %s. Multiple mapping is not supported.", old, exist)
```

Comment:
Updated code change and move this part to separate method CheckStorageClassMapping.
```
func CheckStorageClassMapping(storageClassMapping map[string]string, configData map[string]string, logger logrus.FieldLogger) error {
    for old, new := range configData {
        logger.Infof("Old class name is %s, new class name is %s", old, new)
        if exist, ok := storageClassMapping[old]; ok && exist != "" {
            errMsg := fmt.Sprintf("Old storage class name %s has existing mapping new class storage class name %s.", old, exist)
            logger.Error(errMsg)
            return errors.New(errMsg)
        }
        storageClassMapping[old] = new
    }
    return nil
}
```
configData is retrieved from eligible configMap, and contains all the <old-storage-class>:<new-storage-class> mapping. 
storageClassMapping is empty, in this method it will be filled in. Traverse configData, if old-storage-class appears more than once, return error as we don't support "one old storage class" mapping to multiple new storage class.